src/main/java/com/egen/ordermanagement/service/kafka/ProducerServiceImpl.java

```diff
@@ -0,0 +1,87 @@
+package com.egen.ordermanagement.service.kafka;
+
+import com.egen.ordermanagement.dto.OrderDTO;
+import com.egen.ordermanagement.dto.OrderStatusDTO;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.kafka.support.SendResult;
+import org.springframework.stereotype.Service;
+import org.springframework.util.concurrent.ListenableFuture;
+import org.springframework.util.concurrent.ListenableFutureCallback;
+
+import java.util.List;
+
+@Service
+@Slf4j
+public class ProducerServiceImpl implements ProducerService {
```

Comment:
Good to see that you are exploring and trying stuffs.

Reply:
Thank you gagan. Here I tried implementing a producer produces on to two topics, consumer with one listener consumes from two topics. Both topics with one partition. I am also trying to explore, three consumers from same same group consumes from one topic with 3 partitions. I will let you know if I face any issues.